native/cocos/core/assets/Material.h

```diff
@@ -334,9 +338,33 @@ class Material : public Asset {
             if (macroRecordArray != nullptr) {
                 const auto &patchArray = *macroRecordArray;
                 size_t len = patchArray.size();
-                curr.resize(len);
+                if (curr.size() < len) curr.resize(len);
```

Comment:
The checking of `if (curr.size() < len)` is redundant.
https://en.cppreference.com/w/cpp/container/vector/resize

- Resizes the container to contain count elements, does nothing if count == size().

- If the current size is greater than count, the container is reduced to its first count elements.

- If the current size is less than count,

  - 1) additional [default-inserted](https://en.cppreference.com/w/cpp/named_req/DefaultInsertable) elements are appended.
  - 2) additional copies of value are appended.
```